compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/TypeReference.kt

```diff
@@ -76,14 +77,12 @@ public sealed class TypeReference {
   public fun asTypeNameOrNull(): TypeName? = typeNameOrNull
   public fun asTypeName(): TypeName = typeName
 
-  // Keep this internal, because we need help from other classes for the descriptor
-  // implementation, e.g. FunctionReference resolves the return type or ParameterReference.
-  internal abstract fun resolveGenericTypeOrNull(
-    implementingClass: ClassReference.Psi
+  public abstract fun resolveGenericTypeOrNull(
+    implementingClass: ClassReference
   ): TypeReference?
 
-  internal fun resolveGenericTypeNameOrNull(
-    implementingClass: ClassReference.Psi
+  public fun resolveGenericTypeNameOrNull(
```

Comment:
I think it would make sense to add tests for that in `TypeReferenceTest`

Reply:
Good call, I was thinking this should be covered implicitly but it's probably worth having explicit coverage for this.

Reply:
Tests are pushed now, let me know if it still looks good to you.

Reply:
Tests look great :shipit: